src/microbe_stage/MembraneWaterRipple.cs

```diff
@@ -0,0 +1,572 @@
+        {
+            lastCameraPosition = cameraPos;
+
+            if (parentMembrane != null)
+            {
+                lastCameraDistance = parentMembrane.GlobalPosition.DistanceTo(cameraPos);
+            }
+
+            isCameraPositionValid = true;
+            UpdateLodLevel();
+        }
+    }
+
+    /// <summary>
+    ///   Updates the LOD level based on camera distance
+    /// </summary>
+    private void UpdateLodLevel()
+    {
+        if (!isCameraPositionValid || parentMembrane == null)
+            return;
+
+        LodLevel newLodLevel;
+        float membraneRadius = GetMembraneRadius();
+
+        // Scale the distance thresholds based on the membrane size
+        float sizeScale = Math.Clamp(membraneRadius / 5.0f, 1.0f, 2.0f);
+
+        if (lastCameraDistance > VERY_LOW_LOD_DISTANCE * sizeScale)
+        {
+            newLodLevel = LodLevel.VeryLow;
+        }
+        else if (lastCameraDistance > LOW_LOD_DISTANCE * sizeScale)
+        {
+            newLodLevel = LodLevel.Low;
+        }
+        else if (lastCameraDistance > HIGH_LOD_DISTANCE * sizeScale)
+        {
+            newLodLevel = LodLevel.Medium;
+        }
+        else
+        {
+            newLodLevel = LodLevel.High;
+        }
+
+        // If LOD level changed update subdivision
+        if (newLodLevel != currentLodLevel)
+        {
+            currentLodLevel = newLodLevel;
+            UpdateMeshSubdivision();
+        }
+    }
+
+    /// <summary>
+    ///   Updates mesh subdivision based on current LOD level
+    /// </summary>
+    private void UpdateMeshSubdivision()
+    {
+        if (planeMesh == null)
+            return;
+
+        int subdivision;
+
+        switch (currentLodLevel)
+        {
+            case LodLevel.VeryLow:
+                subdivision = VERY_LOW_LOD_SUBDIVISION;
+                break;
+
+            case LodLevel.Low:
+                subdivision = LOW_LOD_SUBDIVISION;
+                break;
+
+            case LodLevel.Medium:
+                subdivision = MEDIUM_LOD_SUBDIVISION;
+                break;
+
+            case LodLevel.High:
+                subdivision = HIGH_LOD_SUBDIVISION;
+                break;
+
+            default:
+                subdivision = MEDIUM_LOD_SUBDIVISION;
+                break;
+        }
+
+        planeMesh.SubdivideWidth = subdivision;
+        planeMesh.SubdivideDepth = subdivision;
+    }
+
+    /// <summary>
+    ///   Updates the position to exactly match the parent membrane's position
+    /// </summary>
+    private void UpdatePosition()
+    {
+        if (parentMembrane == null || waterPlane == null)
+            return;
+
+        Vector3 membranePos = parentMembrane.GlobalPosition;
+        Vector3 waterPos = membranePos;
+        waterPos.Y += VerticalOffset;
+        waterPlane.GlobalPosition = waterPos;
+
+        // Force water plane to have zero rotation (flat horizontal plane)
+        waterPlane.GlobalRotation = new Vector3(0, 0, 0);
```

Comment:
Setting these separately uses two method calls into the engine. Would be much preferable to just build a Transform and set that with one operation.